### PR TITLE
Added check for eventual symlink when displaying game files.

### PR DESCRIPTION
**Pull request by Shalito01**
Addresses issue #4525. Checks if the file is a symlink it gets the true full path of the file so it shows the correct size.

### CODE REVIEW

**Comment by TSRBerry** [comment]
Could you please add a description and explain why you made this change?

Also you might want to check that commit, it looks like you are using two Github accounts and probably committed with the wrong identity?

File: Ryujinx.Ui.Common/App/ApplicationLibrary.cs [hunk 1]
```diff
@@ -129,13 +129,21 @@ public void LoadApplications(List<string> appDirs, Language desiredTitleLanguage
...
                     }
 
+                    string applicationPath = appPath;
+
+                    // Follows eventual symlink
+                    var tempFile = Directory.ResolveLinkTarget(appPath, true);
+                    if(tempFile != null) {
+                        applicationPath = tempFile.FullName;
+                    }
+
```
Comment (TSRBerry) [inline Ryujinx.Ui.Common/App/ApplicationLibrary.cs:146]: I think we could solve this in a better way if we combine this step with the other operations when we add application paths to the list.

```suggestion
```

File: Ryujinx.Ui.Common/App/ApplicationLibrary.cs [hunk 2]
```diff
@@ -129,13 +129,21 @@ public void LoadApplications(List<string> appDirs, Language desiredTitleLanguage
                 }
 
                 // Loops through applications list, creating a struct and then firing an event containing the struct for each application
-                foreach (string applicationPath in applications)
+                foreach (string appPath in applications)
```
Comment (TSRBerry) [inline Ryujinx.Ui.Common/App/ApplicationLibrary.cs:132]:
```suggestion
                foreach (string applicationPath in applications)
```

File: Ryujinx.Ui.Common/App/ApplicationLibrary.cs [hunk 3]
```diff
@@ -112,9 +112,9 @@ public void LoadApplications(List<string> appDirs, Language desiredTitleLanguage
                             {
                                 return;
                             }
-                        
+
```
Comment (TSRBerry) [inline Ryujinx.Ui.Common/App/ApplicationLibrary.cs:115]:
```suggestion

                            var fileInfo = new FileInfo(app);
```
If we create an instance of `FileInfo` here, we are able to simplify the steps below and check for symlinks as well.

File: Ryujinx.Ui.Common/App/ApplicationLibrary.cs [hunk 4]
```diff
@@ -112,9 +112,9 @@ public void LoadApplications(List<string> appDirs, Language desiredTitleLanguage
...
                             string extension = Path.GetExtension(app).ToLower();
```
Comment (TSRBerry) [inline Ryujinx.Ui.Common/App/ApplicationLibrary.cs:116]:
```suggestion
                            string extension = fileInfo.Extension.ToLower();
```

File: Ryujinx.Ui.Common/App/ApplicationLibrary.cs [hunk 5]
```diff
@@ -112,9 +112,9 @@ public void LoadApplications(List<string> appDirs, Language desiredTitleLanguage
...
-                        
+
                             if (!File.GetAttributes(app).HasFlag(FileAttributes.Hidden) && extension is ".nsp" or ".pfs0" or ".xci" or ".nca" or ".nro" or ".nso")
                             {
                                 applications.Add(app);
```
Comment (TSRBerry) [inline Ryujinx.Ui.Common/App/ApplicationLibrary.cs:120]:
```suggestion
                            if (!fileInfo.Attributes.HasFlag(FileAttributes.Hidden) && extension is ".nsp" or ".pfs0" or ".xci" or ".nca" or ".nro" or ".nso")
                            {
                                var fullPath = fileInfo.ResolveLinkTarget(true)?.FullName ?? fileInfo.FullName;

                                applications.Add(fullPath);
```

Now we only need to add a check for symlinks here and resolve the actual path before we add it to the list of applications.

**Review by TSRBerry — CHANGES_REQUESTED** [review]
I can think we might be able to solve this issue in a better/simpler way.

Let me know what you think about that.

**Comment by Shalito01** [comment]
Yeah, I think it's a lot better in this way.

**Review by riperiperi — APPROVED** [review]
I think this makes sense.

**Review by TSRBerry — APPROVED** [review]
I was waiting for a new review request here, but seems like everything is already done.
Make sure to re-request reviewers once you are done adding changes, so we get a notification to review again.

Lgtm!

**Pull request merged**